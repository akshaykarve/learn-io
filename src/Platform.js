import React from 'react';

const Platform =()=>{
	return(
		<div>
			<img alt='platformImage' src='https://robohash.org/test?200x200'/>
			<div>
				<h3>Tuvan Throat Singing Facts</h3>
			</div>
		</div>
	);
}
export default Platform;