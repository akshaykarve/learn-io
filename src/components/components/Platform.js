import React from 'react';
import '../ComponentStyle.css';

const Platform =({name,platform,setSelectPlatform})=>{
	let showImg;
	if(platform.image===''){
		showImg=<img alt='platformImage' src={`https://robohash.org/${name}?200x200`}/>
	}else{
		showImg=<img alt='platformImage' src={platform.image} height={200} width={200}/>
	}
	
	return(
		<div className='platformStyle grow' onClick={()=>{setSelectPlatform(platform)}}>
			{showImg}
			<div>
				<h4>{name}</h4>
			</div>
		</div>
	);
}

export default Platform;