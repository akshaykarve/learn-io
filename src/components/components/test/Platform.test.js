import React from 'react';
import renderer from 'react-test-renderer';
import {default as Platform} from "../Platform";

describe("Single Platform Component", ()=>{
	it('no name', () => {
		
		const tree = renderer.create(
			<Platform name={''}
			/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('test name', () => {
		
		const tree = renderer.create(
			<Platform name={'test'}
			/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});
});