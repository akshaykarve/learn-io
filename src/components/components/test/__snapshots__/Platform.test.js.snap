// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single Platform Component no name 1`] = `
<div
  className="platformStyle grow"
  onClick={[Function]}
>
  <img
    alt="platformImage"
    src="https://robohash.org/?200x200"
  />
  <div>
    <h4>
      
    </h4>
  </div>
</div>
`;

exports[`Single Platform Component test name 1`] = `
<div
  className="platformStyle grow"
  onClick={[Function]}
>
  <img
    alt="platformImage"
    src="https://robohash.org/test?200x200"
  />
  <div>
    <h4>
      test
    </h4>
  </div>
</div>
`;
